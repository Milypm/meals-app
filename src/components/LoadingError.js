import FadeLoader from 'react-spinners/FadeLoader';
import { css } from '@emotion/react';
import '../styles/home.css';

const override = css`
  display: block;
  margin: 0 auto;
  border-color: red;
`;

export const Loading = () => {
  const loading = true;
  return (
    <div className="loading-div">
      <h3>Loading</h3>
      <FadeLoader color="rgb(219, 135, 96)" loading={loading} css={override} size={850} />
    </div>
  );
};
export const Error = () => (
  <div className="error-div">
    <h3>Sorry, we couldn&apos;t display the information :( Try again please</h3>
  </div>
);
