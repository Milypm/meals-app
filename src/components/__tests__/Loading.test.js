import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Loading } from '../LoadingError';

describe('Loading and Error components', () => {
  test('Loading component renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Loading />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
