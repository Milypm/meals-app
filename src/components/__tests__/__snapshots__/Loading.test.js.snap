// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading and Error components Loading component renders correctly 1`] = `
<div
  className="loading-div"
>
  <h3>
    Loading
  </h3>
  <span
    className="css-8gz6np"
  >
    <span
      className="css-xya1tf"
    />
    <span
      className="css-v4vq26"
    />
    <span
      className="css-kqipwe"
    />
    <span
      className="css-def0y6"
    />
    <span
      className="css-1cfbd8g"
    />
    <span
      className="css-w0j3t9"
    />
    <span
      className="css-h7en48"
    />
    <span
      className="css-mg8it5"
    />
  </span>
</div>
`;
